src/clients/detector/text_context_doc.rs

```diff
@@ -0,0 +1,112 @@
+use async_trait::async_trait;
+use hyper::{HeaderMap, StatusCode};
+use serde::{Deserialize, Serialize};
+
+use super::{DetectorError, DETECTOR_ID_HEADER_NAME};
+use crate::{
+    clients::{Client, Error, HttpClient},
+    health::HealthCheckResult,
+    models::{DetectionResult, DetectorParams},
+};
+
+#[cfg_attr(test, faux::create)]
+#[derive(Clone)]
+pub struct TextContextDocDetectorClient {
+    client: HttpClient,
+}
+
+#[cfg_attr(test, faux::methods)]
+impl TextContextDocDetectorClient {
+    pub fn new(client: HttpClient) -> Self {
+        Self { client }
+    }
+
+    pub async fn text_context_doc(
+        &self,
+        model_id: &str,
+        request: ContextDocsDetectionRequest,
+        headers: HeaderMap,
+    ) -> Result<Vec<DetectionResult>, Error> {
+        let url = self
+            .client
+            .base_url()
+            .join("/api/v1/text/context/doc")
+            .unwrap();
+        let response = self
+            .client
+            .post(url)
+            .headers(headers)
+            .header(DETECTOR_ID_HEADER_NAME, model_id)
+            .json(&request)
+            .send()
+            .await?;
+        if response.status() == StatusCode::OK {
+            Ok(response.json().await?)
+        } else {
+            let code = response.status().as_u16();
+            let error = response
+                .json::<DetectorError>()
+                .await
+                .unwrap_or(DetectorError {
+                    code,
+                    message: "".into(),
+                });
+            Err(error.into())
+        }
+    }
+}
```

Comment:
This method is quite similar in all detector clients (it basically invokes the detector and parse the response). The differences are the URL and the request/response types.

Should we create a default implementation for it using generics? Or maybe use some kind of template method? Or even create a helper function to abstract away repetitive code?

Note: definitely not something that needs to be done in this PR, I just wanted to start the discussion.

Reply:
I actually initially created a helper function used by each implementation to reduce this boilerplate, but then I took it out in favor of keeping things explicit and I didn't want to assume that all detector implementations will have the exact same error format, header requirements, etc. I can add it back if you guys would prefer that.

Reply:
I don't really have a strong opinion on this one. @evaline-ju, @gkumbhat, any thoughts?

Reply:
Actually being explicit might work out for us better, given the variability of detectors and even implementation of detectors. This would provide us a way to handle differences more clearly at cost of some duplication, but I think thats okay.